lib/bloc/user_profile/fiat_currency.dart

```diff
@@ -0,0 +1,46 @@
+enum FiatCurrencyID { USD, EUR, GBP, JPY }
+
+class FiatCurrency extends Object {
+  final String symbol;
+  static const FiatCurrency USD = FiatCurrency._internal("\$");
+  static const FiatCurrency EUR = FiatCurrency._internal("€");
+  static const FiatCurrency GBP = FiatCurrency._internal("£");
+  static const FiatCurrency JPY = FiatCurrency._internal("¥");
+  static final List<FiatCurrency> currencies = List.unmodifiable([USD, EUR, GBP, JPY]);
+
+  const FiatCurrency._internal(this.symbol);
+
+  factory FiatCurrency.fromSymbol(String symbol) {
+    return currencies.firstWhere((c) => c.shortName == symbol);
+  }
+
+  String get shortName {
+    switch (symbol) {
+      case "€":
+        return "EUR";
+      case "£":
+        return "GBP";
+      case "¥":
+        return "JPY";
+      case "\$":
+        return "USD";
+      default:
+        return "USD";
+    }
+  }
+
+  String get logoPath {
```

Comment:
Is this really a sustainable solution if we wish to support many fiat currencies? Do we really need a separate icon for each currency? Most fiat currencies have a unicode symbol, perhaps we could rely on those and avoid creating new icons for each currency? I think it would be great if the fiat conversion could be general enough so that more currencies could be added without much work.

Reply:
@jemilsson good point!
@erdemyerebasmaz my suggestion is to use a data file (see this for example: https://developers.google.com/public-data/docs/canonical/currencies_csv) to represents all currencies.. My first approach would be to look for this data in dart/flutter but I couldn't find a built-in feature that supports both currency short name and symbol. Anyway we can change the data representation later if built in support is added. The important thing is that these currencies are dynamically loaded and future support for additional currencies won't require code changes.
As for the icon let's talk to Roy about his view before any changes.